go/vt/mysqlctl/mysqld.go

```diff
@@ -1387,7 +1392,7 @@ func (mysqld *Mysqld) ReadBinlogFilesTimestamps(ctx context.Context, req *mysqlc
 			return matchedTime, false, err
 		}
 		go scan()
-		if err := mysqlbinlogCmd.Wait(); err != nil {
+		if err := mysqlbinlogCmd.Wait(); err != nil && !intentionalKill {
```

Comment:
Again, not strictly part of the purpose of the PR, but I noticed a scenario where the backup process would hang. This mitigates the scenario.